cmd/util/cmd/epoch-recovery/cmd/reset.go

```diff
@@ -0,0 +1,192 @@
+package cmd
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+
+	"github.com/onflow/cadence"
+
+	jsoncdc "github.com/onflow/cadence/encoding/json"
+
+	"github.com/onflow/flow-go/engine/common/rpc/convert"
+	"github.com/onflow/flow-go/model/bootstrap"
+	"github.com/onflow/flow-go/state/protocol/inmem"
+	"github.com/onflow/flow-go/utils/io"
+)
+
+var (
+	flagBootDir string
+	flagPayout  uint64
+)
+
+// resetCmd represents a command to reset epoch data in the Epoch smart contract
+var resetCmd = &cobra.Command{
+	Use:   "reset",
+	Short: "Generates `resetEpoch` JSON transaction arguments.",
+	Long:  "Generates `resetEpoch` transaction arguments from a root protocol state snapshot and writes it to a JSON file",
+	Run:   resetRun,
+}
+
+func init() {
+	rootCmd.AddCommand(resetCmd)
+	addResetCmdFlags()
+}
+
+func addResetCmdFlags() {
+	resetCmd.Flags().StringVar(&flagBootDir, "boot-dir", "", "path to the directory containing the bootstrap files")
+	_ = resetCmd.MarkFlagRequired("boot-dir")
+
+	resetCmd.Flags().Uint64Var(&flagPayout, "payout", 0, "the payout")
+}
+
+// resetRun resets epoch data in the Epoch smart contract with fields generated
+// from the root-protocol-snapshot.json
+func resetRun(cmd *cobra.Command, args []string) {
```

Comment:
For testing the new code, let's use the same pattern as [previous CLI changes](https://github.com/onflow/flow-go/blob/d615293d74ecf6438d71bd0a95733ae0d9611f4c/Makefile#L385-L404).

We can:
* Generate a root state snapshot and write it to disk
* Execute this command (with and without the payout flag set)
* Read the resulting arguments file, validating that the read inputs match those from the root state snapshot (and `--payout` flag)

Reply:
Sounds like a plan!